feat: Add a Chapter on Accessibility in the Rustdoc book (#151106)
rust-lang/rust: src/doc/rustdoc/src/accessibility.md

# A Note on Accessibility in Rustdoc

`rustdoc` is WCAG-compliant[^1] and implements accessible documentation generation, with support for many accessibility practices and we strive to fully comply with WCAG Level AA[^2].

## Accessibility Practices Followed by Rustdoc

1. `rustdoc` only uses JavaScript to make your experience better, but `rustdoc` works perfectly well without it, meaning that most interactive features should work well without JavaScript. This ensures that no hidden scripts or interactive elements mess with screen readers and other accessibility tools.
2. `rustdoc` uses semantic HTML elements wherever possible.
    1. We use `<details>` instead of checkboxes so text can still be searched in hidden and collapsed content.

>     2. We use the `title` attribute to make your experience better, but `rustdoc` is perfectly usable without them. The **`title`** global attribute contains text representing advisory information related to the element it belongs to.[^4]

Reviewer: ```suggestion
    2. We use the `title` attribute on HTML elements. The **`title`** attribute contains text representing advisory information related to the element it belongs to.[^4]
```

Also it's the only attribute/tag you explain.

Author: Yeah I need to expand on this a bit more, haven't gotten enough time though rn due to exams and all

Reviewer 2: Isn't the whole point of this note to elaborate that `title` should never be used for vital functionality?
